packages/config-service/src/services/globalConfig.ts

```diff
@@ -26,7 +28,7 @@ export interface ConfigProperty {
 }
 
 export class GlobalConfig {
-  public static readonly ENTRIES: Record<string, ConfigProperty> = {
+  public static readonly ENTRIES: Record<ConfigName, ConfigProperty> = {
```

Comment:
Instead of enumerating the keys manually, can we do something like
```typescript

const _CONFIG = {
  BATCH_REQUESTS_ENABLED: {
    envName: 'BATCH_REQUESTS_ENABLED',
    type: 'boolean',
    required: false,
    defaultValue: null,
  },
  //...
  WS_SUBSCRIPTION_LIMIT: {
    envName: 'WS_SUBSCRIPTION_LIMIT',
    type: 'number',
    required: false,
    defaultValue: null,
  },
} satisfies { [key: string]: ConfigProperty };

export type ConfigKey = keyof typeof _CONFIG;

export class GlobalConfig {
  public static readonly ENTRIES: Record<ConfigKey, ConfigProperty> = _CONFIG;
}

```
?

So we retain type safety for the keys (even if they're strings).